### PR TITLE
feat: bold

### PR DESCRIPTION
**Pull request by gzeoneth**
merging https://github.com/OffchainLabs/bold/commit/6b42a38f /contracts into `nitro-contracts`

contracts should be identical to the https://github.com/OffchainLabs/bold except some mocks, precompiles interface, and some renamed variables in BOLDUpgradeAction
```
$ diff -qr ./src ../bold/contracts/src
Only in ./src: chain
Only in ./src/mocks: Benchmarks.sol
Only in ./src/mocks: MultiCallTest.sol
Only in ./src/mocks: PendingBlkTimeAndNrAdvanceCheck.sol
Only in ./src/mocks: Program.sol
Only in ./src/mocks: SdkStorage.sol
Only in ./src/mocks: SimpleCacheManager.sol
Files ./src/precompiles/ArbWasmCache.sol and ../bold/contracts/src/precompiles/ArbWasmCache.sol differ
Files ./src/rollup/BOLDUpgradeAction.sol and ../bold/contracts/src/rollup/BOLDUpgradeAction.sol differ
```

e2e test is currently disabled due to lack of nitro support

resolves BLK-176

### CODE REVIEW

**Comment by gzeoneth** [comment]
https://github.com/OffchainLabs/nitro-contracts/pull/196
thoughts on bumping all npm packages in nitro-contracts to latest?

**Review by yahgwai — APPROVED** [review]

File: src/rollup/BOLDUpgradeAction.sol [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+            delaySeconds / SECONDS_PER_SLOT + 1;
+
+        delayBlocks = implDelayBlocks > delayBlocks ? implDelayBlocks : delayBlocks;
+
+        ISequencerInbox(SEQ_INBOX).setMaxTimeVariation(ISequencerInbox.MaxTimeVariation({
+            delayBlocks: delayBlocks,
+            delaySeconds: delaySeconds,
+            futureBlocks: futureBlocks,
+            futureSeconds: futureSeconds
+        }));
```
Comment (gzeoneth) [inline src/rollup/BOLDUpgradeAction.sol:487]: we probably should also update futureBlocks to align

Comment (gzeoneth) [reply src/rollup/BOLDUpgradeAction.sol:489]: BLK-246

**Review by godzillaba — DISMISSED** [review]
:tada:

**Review by yahgwai — APPROVED** [review]

**Pull request merged**